### ISSUE TITLE
java: Parameter keys are not escaped properly

### ISSUE THREAD

**Issue by lorenzleutgeb** [OpenAPITools/openapi-generator#459]
##### Description

Note the `|` character in the parameter key. This translates to:

```java
  public static final String SERIALIZED_NAME_UPDATED_AT|GT = "updated_at|gt";
  @SerializedName(SERIALIZED_NAME_UPDATED_AT|GT)
  private OffsetDateTime updatedAt|gt = null;
```

... which does not parse as Java.

##### openapi-generator version

020883fd4da67050a6ae1b228daa8cef6849d732

##### OpenAPI Declaration

```yaml
paths:
  '/foo':
    get:
      parameters:
        - name: filter
          in: query
          required: false
          style: deepObject
          schema:
            properties:
              "updated_at|gt":
                type: string
                format: date-time
```

##### Command Line used for generation

```
java -jar openapi-generator-cli.jar generate -l java -i api.yml -o jclient
```

##### Steps to reproduce

Generate client.

##### Related issues/PRs

None.

##### Suggest a fix/enhancement

Escape parameter names properly such that the generator emits syntactically valid Java code.

**Comment by jmini** [comment]
Thank you a lot for this issue. Have you the possibility to check the patch I have proposed in #461?

**Comment by lorenzleutgeb** [comment]
Yes, I checked and the fix works for me. Thank you!

**Comment by jmini** [comment]
This is fixed in `3.1.0-SNAPSHOT` (release is planned tomorrow)
